src/components/interaction-designer/BuilderCanvas.vue

```diff
@@ -42,27 +45,23 @@ export class BuilderCanvas extends Vue {
   // ###### Validation API Watchers [
   @Watch('activeFlow', {deep: true, immediate: true})
   async onActiveFlowChanged(newFlow: IFlow): Promise<void> {
-    console.debug('watch/activeFlow:', 'active flow has changed from builder canvas, validating ...')
-    await this.validate_flow({flow: newFlow})
+    this.debounceFlowValidation({newFlow})
   }
 
+  debounceFlowValidation = debounce(function (this: any, {newFlow}: {newFlow: IFlow}) {
+    console.debug('watch/activeFlow:', 'active flow has changed from builder canvas, validating ...')
+    this.validate_flow({flow: newFlow})
+  }, DEBOUNCE_VALIDATION_TIMER)
+
   @Watch('blocksOnActiveFlowForWatcher', {deep: true, immediate: true})
   async onBlocksInActiveFlowChanged(newBlocks: IBlock[], oldBlocks: IBlock[]): Promise<void> {
-    if (newBlocks.length === 0) {
-      return
-    }
+    this.debounceBlockValidation()
+  }
 
+  debounceBlockValidation = debounce(function (this: any) {
     console.debug('watch/activeFlow.blocks:', 'blocks inside active flow have changed, validating ...')
-
-    await Promise.all(
-      newBlocks.map(async (currentNewBlock) => {
-        const currentOldBlock = find(oldBlocks, {uuid: currentNewBlock.uuid})
-        if (!isEqual(currentNewBlock, currentOldBlock)) {
-          await this.validate_block({block: currentNewBlock})
-        }
-      }),
-    )
-  }
+    this.validate_allBlocksWithinFlow()
+  }, DEBOUNCE_VALIDATION_TIMER)
```

Comment:
For blocks I've changed it so that we are validating not only the blocks that have changed, but all blocks.
We need it because if the changed (e.g. renamed) block is referenced in a different block (e.g. in date expression) we want to invalidate that.